libyul/AsmParser.cpp

```diff
@@ -78,14 +81,70 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+langutil::Token Parser::advance()
+{
+	auto const token = ParserBase::advance();
+	if (m_sourceMap)
+		fetchSourceLocationFromComment();
+	return token;
+}
+
+void Parser::fetchSourceLocationFromComment()
+{
+	solAssert(m_sourceMap.has_value(), "");
+
+	if (m_scanner->currentCommentLiteral().empty())
+		return;
+
+	static std::regex const lineRE = std::regex(
+		R"~~~(\s*@src\s+(\d+):(-1|\d+):(-1|\d+)\s*)~~~",
```

Comment:
This still succeeds for `x@src 1:1:1`

Reply:
ah, due to the missing leading `^` which I removed earlier.

Reply:
`@a @src 1:1:1` should also be valid.

Reply:
so `.*\s@src......`? that should consume anything ahead.